File: go/workflows/shipping_child_workflow.go
package workflows

import (
	"temporal-order-management/activities"
	"temporal-order-management/app"
	"time"

	"go.temporal.io/sdk/temporal"
	"go.temporal.io/sdk/workflow"
)

func ShippingWorkflow(ctx workflow.Context, input app.ShippingInput) (string, error) {
	logger := workflow.GetLogger(ctx)
	logger.Info("Shipping workflow started", "orderId", input.Order.OrderId)

	activityOptions := workflow.ActivityOptions{
		StartToCloseTimeout: 5 * time.Second,
		RetryPolicy: &temporal.RetryPolicy{
			InitialInterval:    1 * time.Second,
			BackoffCoefficient: 2.0,
			MaximumInterval:    30 * time.Second,
		},
	}
	ctx = workflow.WithActivityOptions(ctx, activityOptions)

	err := workflow.ExecuteActivity(ctx, activities.ShipOrder, input).Get(ctx, nil)
	if err != nil {
		return "", err
	}

	return "", nil
}

File: ruby/workflows/shipping_child_workflow.rb
require 'temporalio/workflow'
require_relative '../shared_objects'

module Workflows
  class ShippingChildWorkflow < Temporalio::Workflow::Definition
    attr_accessor :retry_policy

    def initialize
      @retry_policy = nil
    end

    def execute(input, item)
      logger.info("Shipping workflow started, orderId = #{input.order_id}")

      Temporalio::Workflow.execute_activity(
        Activities::ShipOrderActivity,
        input, item,
        start_to_close_timeout: 5,
        retry_policy: @retry_policy
      )
    end

    private

    def logger
      @logger ||= Temporalio::Workflow.logger
    end
  end
end
